shell/shared/grain.js

```diff
@@ -411,7 +411,11 @@ Router.map(function () {
         result.appOrigin = document.location.protocol + "//" + makeWildcardHost(session.hostId);
         setCurrentSessionId(session._id, result.appOrigin, grainId);
         result.sessionId = session._id;
-        result.path = encodeURIComponent("/" + (self.params.path || ""));
+        var currentPath = Iron.Location.get().path;
```

Comment:
For what it's worth:

https://bugzilla.mozilla.org/show_bug.cgi?id=483304

`window.location.hash` differences between Firefox and everyone else.